### PR TITLE
pubish umd to npm mk II

### PR DESCRIPTION
**Pull request by graingert**
This doesn't include the manual media changes

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -47,12 +47,16 @@ script:
...
-  password:
-    secure: fEP9K7y0ZF9fRvQEUN4kgPXQEZvi3Cx3ikUebG2UM/2uhcaUQm0+KpgZ2S+lvOTYcBnNgzPzFsVIZMcVcTxwIKuQDEMq9y2eop2hnisb9KXsIm9qPYSzOnRm74VuiOt4hNOZMe0qVBK2cO3vC9NDXuzdI8A0JynJhthfl4t+kFM=
+  - provider: pypi
+    distributions: sdist bdist_wheel
+    user: codingjoe
+    password:
+      secure: fEP9K7y0ZF9fRvQEUN4kgPXQEZvi3Cx3ikUebG2UM/2uhcaUQm0+KpgZ2S+lvOTYcBnNgzPzFsVIZMcVcTxwIKuQDEMq9y2eop2hnisb9KXsIm9qPYSzOnRm74VuiOt4hNOZMe0qVBK2cO3vC9NDXuzdI8A0JynJhthfl4t+kFM=
+  - provider: npm
+    email: info@johanneshoppe.com
+    api_key: YOUR_SECURE_API_KEY_HERE
```
Comment (graingert) [inline .travis.yml]: @codingjoe you'll need to add your `npm` credentials here

Comment (codingjoe) [reply .travis.yml]: I am pretty sure you need to duplicate the `on:` section for both providers.

Comment (codingjoe) [reply .travis.yml]: I'll add a key when I merge.

File: django_select2/static/django_select2/django_select2.js [hunk 2]
```diff
@@ -1,4 +1,15 @@
-(function ($) {
+/* global define, jQuery */
+(function (factory) {
```
Comment (codingjoe) [inline django_select2/static/django_select2/django_select2.js]: I am not so firm with modern day JavaScript. Why do we need all this boilerplate?

File: set_version.py [hunk 3]
```diff
@@ -0,0 +1,11 @@
+#!/usr/bin/env python3
```
Comment (codingjoe) [inline set_version.py]: Is this file omited in the setuptools package?

File: set_version.py [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+import json
+import os
+
+def main():
+    with open('package.json', 'r+') as f:
```
Comment (codingjoe) [inline set_version.py]: You should open this writable, should you not?

File: set_version.py (hunk 3 above)
Comment (graingert) [reply set_version.py]: Should be

File: set_version.py (hunk 4 above)
Comment (graingert) [reply set_version.py]: That's what 'r+' means I'm told

File: django_select2/static/django_select2/django_select2.js (hunk 2 above)
Comment (graingert) [reply django_select2/static/django_select2/django_select2.js]: It's one for amd, one for commonjs and one for browser globals. It's known as UMD

File: set_version.py (hunk 4 above)
Comment (codingjoe) [reply set_version.py]: Yes, sorry. You need read/write, makes sense.

File: django_select2/static/django_select2/django_select2.js (hunk 2 above)
Comment (codingjoe) [reply django_select2/static/django_select2/django_select2.js]: You gotta give me more. Consider me a JS noob. Is that really needed or just a nice to have?

I really want to keep this simple... I guess you know why ;)

File: set_version.py (hunk 3 above)
Comment (codingjoe) [reply set_version.py]: haha, I'll double check when I merge ;)

File: django_select2/static/django_select2/django_select2.js (hunk 2 above)
Comment (graingert) [reply django_select2/static/django_select2/django_select2.js]: it's so it works with `webpack`, `require.js` and as a django static asset

File: .travis.yml (hunk 1 above)
Comment (graingert) [reply .travis.yml]: I'm not sure

File: django_select2/static/django_select2/django_select2.js (hunk 2 above)
Comment (codingjoe) [reply django_select2/static/django_select2/django_select2.js]: Can you add a comment where the wrapper snippet is coming from. That way future contributors will know whats up.

**Comment by graingert** [comment]
@codingjoe any update on this?

**Comment by codingjoe** [comment]
@graingert did you give me access yet for the npm package?

**Comment by codingjoe** [comment]
@graingert `codingjoe` ;)

**Comment by graingert** [comment]
@codingjoe any update on this?

**Comment by codingjoe** [comment]
Travis is currently not working. I will tag this once it's working again and check if it releases correctly. If I forget please ping me again ;)

**Comment by graingert** [comment]
@codingjoe awesome thanks

**Comment by codingjoe** [comment]
Ah your commit actually broke the travis config. I missed that. I should be working now. I am just waiting for the deploy to succeed.

**Comment by codingjoe** [comment]
Released in 6.3.0

**Comment by codingjoe** [comment]
ok @graingert, I think you will need to put a bit more love into this
https://travis-ci.org/applegrew/django-select2/jobs/419135968

**Comment by graingert** [comment]
Looks like you need to 'skip_cleanup' to deploy build artifacts:

Cleaning up git repository with `git stash --all`. If you need build
artifacts for deployment, set `deploy.skip_cleanup: true`. See
https://docs.travis-ci.com/user/deployment#Uploading-Files-and-skip_cleanup.

**Comment by codingjoe** [comment]
@graingert I can't wait for your pull-request 👍

**Pull request merged**